src/Core/Managed/Shared/Extensibility/MetricManager.cs

```diff
@@ -0,0 +1,285 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+
+#if CORE_PCL || NET45 || NET46
+    using TaskEx = System.Threading.Tasks.Task;
+#endif
+
+    /// <summary>
+    /// Metric factory and controller.
+    /// </summary>
+    public sealed class MetricManager : IDisposable
+    {
```

Comment:
Looking further, we throw in disposed state for methods, but not properties. Is this right. Most methods, actually, would work fine in the disposed state, the only thing we cannot do is schedule future updates. Can we allow those calls after Dispose()? Can we get rid of IDisposable? I think we should chat. :)